poncho/src/poncho/library_network_code.py

```diff
@@ -145,10 +157,27 @@ def start_function(in_pipe_fd, thread_limit=1):
                         event = cloudpickle.load(f)
 
                     # output of execution should be dumped to outfile.
-                    with open('outfile', 'wb') as f:
-                        cloudpickle.dump(globals()[function_name](event), f)
+                    exit_status = 0
+                    try:
+                        result = globals()[function_name](event)
+                        try:
+                            with open('outfile', 'wb') as f:
+                                cloudpickle.dump(result, f)
+                        except Exception:
+                            if os.path.exits('outfile'):
+                                os.remove('outfile')
+                            raise
+                    except Exception:
+                        exit_status = 1
+
+                    try:
+                        if not result["Success"]:
+                            exit_status = 1
+                    except Exception:
+                        exit_status = 1
```

Comment:
Maybe a different exit code here?

Reply:
Exit code only means call failed. The actual error is in event if it was created at all. This exit code is just checked by the manager to be nonzero. In regular tasks the exit code is checked by the user, but here there is no user...